query/physicalplan/aggregate.go

```diff
@@ -315,14 +320,25 @@ func (a *HashAggregate) Callback(ctx context.Context, r arrow.Record) error {
 			}
 		}
 
-		if a.columnToAggregate.MatchColumn(field.Name) {
-			columnToAggregate = r.Column(i)
-			aggregateFieldFound = true
+		for j, col := range a.columns {
+			if a.finalStage {
+				if col.resultName == field.Name {
+					columnToAggregate[j] = r.Column(i)
+					aggregateFieldFound[j] = true
+				}
+			} else {
+				if col.expr.MatchColumn(field.Name) {
+					columnToAggregate[j] = r.Column(i)
+					aggregateFieldFound[j] = true
+				}
+			}
```

Comment:
What was the need for the addition of these `if` statements? If they're necessary I would add a comment and here's a rewrite suggestion:
```suggestion
				if col.expr.MatchColumn(field.Name) || col.resultName == field.Name {
					columnToAggregate[j] = r.Column(i)
					aggregateFieldFound[j] = true
				}
```